### PR TITLE
MSI: RHODS-RHOAM Co-exist

### PR DESCRIPTION
**Pull request by oharan2**

### CODE REVIEW

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+base_images:
+  cli:
+    name: "4.13"
+    namespace: ocp
+    tag: cli
+  ocp-addons-operators-cli:
+    name: ocp-addons-operators-cli
+    namespace: ci
+    tag: latest
```
Comment (omrirh) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml:9]: you will need to add `ods-ci` image mirror declaration to base_images

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
Depends on #40346

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: rhods-rhoam-tests
+  cron: 0 12 * * 6
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml:25]: the job will be triggred when a new rhods or rhoam addon is released; plese update `cron` to run on a yearly basis.

File: ci-operator/step-registry/addon/install/addon-install-commands.sh [hunk 3]
```diff
@@ -9,15 +9,17 @@ CLUSTER_NAME=$(cat "${SHARED_DIR}/cluster-name")
...
 ADDONS_CMD=""
-for addon_value in $(env | grep -E '^ADDON[0-9]+_CONFIG'); do
+for addon_value in $(env | grep -E '^ADDON[0-9]+_CONFIG' | sort  --version-sort); do
     addon_value=$(echo "$addon_value" | sed -E  's/^ADDON[0-9]+_CONFIG=//')
     if  [ "${addon_value}" ]; then
       ADDONS_CMD+=" --addon ${addon_value} "
     fi
 done
-
+echo "ADDONS_CMD: ${ADDONS_CMD}"
```
Comment (rnetser) [inline ci-operator/step-registry/addon/install/addon-install-commands.sh:22]: can you explain what is the added value? we have  the full command echo.

File: ci-operator/step-registry/addon/rosa/addon-rosa-workflow.yaml [hunk 4]
```diff
@@ -8,7 +8,6 @@ workflow:
       ENABLE_AUTOSCALING: "false"
       REPLICAS: "2"
       MULTI_AZ: "false"
-      PARALLEL: "true"
     post:
```
Comment (rnetser) [inline ci-operator/step-registry/addon/rosa/addon-rosa-workflow.yaml:11]: please re-add `PARALLEL: "true"` - this is the default we'd like in most cases; set to "" in config file.

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      version: "4.13"
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: rhods-rhoam-tests
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml:29]: please rename the test
see https://docs.ci.openshift.org/docs/how-tos/naming-your-ci-jobs/ and the naming of the team's tests

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml (hunk 5 above)
Comment (oharan2) [reply ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml]: is `mps-pipeline-rosa-rhods-rhoam` legitimate?

`MPS-Pipeline` is a suitable test marker for rhods-rhoam co-exist tests. 
The equivalence marker is `RHOAM-RHODS` which would have made duplication with the naming scheme 
..v4.13-[tests suit]-[platform]-[addons installed]:
`rhoam-rhoads-rosa-rhods-rhoam`

Should `mps-pipeline` replaced with `sanity`?

Reference:
https://github.com/red-hat-data-services/ods-ci/blob/478b497fbde17483b59c88b4dbb9faa9c15fe5ae/ods_ci/tests/Tests/600__ai_apps/640__rhoam/640__rhoam.robot#L23C15-L23C27

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-rhods-rhoam-tests

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-mps-pipeline-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-mps-pipeline-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-mps-pipeline-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-co-exist-rosa-rhods-rhoam

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+tests:
+- as: co-exist-rosa-rhods-rhoam
+  cron: '@yearly'
+  steps:
+    cluster_profile: aws-managed-cspi-qe
+    env:
+      ADDON1_CONFIG: name=managed-odh;notification-email=interop-qe-ms@redhat.com
+      ADDON2_CONFIG: name=managed-api-service;addon-managed-api-service=1;rosa-cli-required=true;notification-email=interop-qe-ms@redhat.com;rosa=true
+      OPENSHIFT_VERSION: "4.13"
+      PARALLEL: ""
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml:37]: please use `"false"` (updated the logic to be more strict in https://github.com/openshift/release/pull/40961)

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml (hunk 6 above)
Comment (oharan2) [reply ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-rhoam-addons__v4.13.yaml]: Thanks

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-addons-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-v4.13-co-exist-rosa-rhods-rhoam

**Comment by oharan2** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-rhoam-v4.13-co-exist-rosa-rhods-rhoam

**Comment by rnetser** [comment]
please rebase on https://github.com/openshift/release/pull/41075

**Review by myakove — APPROVED** [review]

**Review by omrirh — APPROVED** [review]

**Pull request merged**